src/poetry/console/commands/new.py

```diff
@@ -71,8 +70,16 @@ def handle(self) -> int:
             if author_email:
                 author += f" <{author_email}>"
 
-        current_env = SystemEnv(Path(sys.executable))
-        default_python = "^" + ".".join(str(v) for v in current_env.version_info[:2])
+        poetry_config = Config.create()
+        default_python = "^" + ".".join(
+            str(v)
+            for v in EnvManager.get_python_version(
+                prefer_active_python=poetry_config.get(
+                    "virtualenvs.prefer-active-python"
+                ),
+                io=self.io,
+            )[:2]
+        )
```

Comment:
Could you put it into a utils function? we could reduce duplication (command classes are already a mess)

Reply:
I refactored it a bit, so this splitting and joining stuff happens in `EnvManager.get_python_version`.

Reply:
- [Python] -